src/main/java/seedu/address/logic/commands/AddCommand.java

```diff
@@ -47,14 +48,30 @@ public AddCommand(Module module) {
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
 
+        // Database check
+        if (!model.checkDbValidModule(toAdd)) {
+            throw new CommandException(
+                    String.format(Messages.MESSAGE_INVALID_MODULE_CODE, toAdd.getModuleCode()));
+        }
 
+        // Module plan duplicate check
         if (model.hasModule(toAdd)) {
             throw new CommandException(MESSAGE_DUPLICATE_MODULE);
-
         }
 
-        model.addModule(toAdd);
-        return new CommandResult(String.format(MESSAGE_SUCCESS, Messages.format(toAdd)));
+        // Populating module details from database
+        ModuleCode moduleCode = toAdd.getModuleCode();
+        Module newModule = new Module(moduleCode,
+                                      toAdd.getYearTaken(),
+                                      toAdd.getSemesterTaken(),
+                                      toAdd.getGrade(),
+                                      model.getDbModuleName(moduleCode),
```

Comment:
Maybe we should move this to Module constructor

Reply:
I was considering that, but I feel a better solution would be to remove DbModule entirely. 

The idea is to have ModuleData create all possible Module classes (instead of DbModule) with the proper ModuleCode, ModuleName, ModularCredits and ModuleDescription upon application launch. 

Then, instead of creating Modules, AddCommand/AddCommandParser will be retrieving from ModuleData based on the user input ModuleCode, and then filling in the other user input fields through some method like `fillUserInputs(Year, Semester, Grade)` instead.

Similarly, the DeleteCommand would just call some method `clearUserInputs()` to return the Module to its original state.

But this would require some refactoring, so I wanted to consult with everyone first and just left the above as a temporary working solution.